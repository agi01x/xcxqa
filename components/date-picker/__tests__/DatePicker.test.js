import React from 'react';
import { mount } from 'enzyme';
import moment from 'moment';
import MockDate from 'mockdate';
import DatePicker from '..';
import {
  selectDate,
  openPanel,
  clearInput,
  nextYear,
  nextMonth,
  hasSelected,
} from './utils';
import focusTest from '../../../tests/shared/focusTest';

describe('DatePicker', () => {
  focusTest(DatePicker);

  beforeEach(() => {
    MockDate.set(moment('2016-11-22'));
  });

  afterEach(() => {
    MockDate.reset();
  });

  it('prop locale should works', () => {
    const locale = {
      lang: {
        placeholder: 'Избери дата',
        rangePlaceholder: [
          'Начална дата',
          'Крайна дата',
        ],
        today: 'Днес',
        now: 'Сега',
        backToToday: 'Към днес',
        ok: 'Добре',
        clear: 'Изчистване',
        month: 'Месец',
        year: 'Година',
        timeSelect: 'Избор на час',
        dateSelect: 'Избор на дата',
        monthSelect: 'Избор на месец',
        yearSelect: 'Избор на година',
        decadeSelect: 'Десетилетие',
        previousMonth: 'Предишен месец (PageUp)',
        nextMonth: 'Следващ месец (PageDown)',
        previousYear: 'Последна година (Control + left)',
        nextYear: 'Следваща година (Control + right)',
        previousDecade: 'Предишно десетилетие',
        nextDecade: 'Следващо десетилетие',
        previousCentury: 'Последен век',
        nextCentury: 'Следващ век',
        yearFormat: 'YYYY',
        dateFormat: 'D M YYYY',
        dayFormat: 'D',
        dateTimeFormat: 'D M YYYY HH:mm:ss',
        monthBeforeYear: true,
      },
      timePickerLocale: {
        placeholder: 'Избор на час',
      },
    };
    const birthday = moment('2000-01-01', 'YYYY-MM-DD');
    const wrapper = mount(
      <DatePicker open locale={locale} value={birthday} />
    );
    expect(wrapper.render()).toMatchSnapshot();
  });

  // Fix https://github.com/ant-design/ant-design/issues/8885
  it('control value after panel closed', () => {
    class Test extends React.Component {
      state = {
        cleared: false,
        value: moment(),
      }

      onChange = (value) => {
        let { cleared } = this.state;

        if (cleared) {
          value = moment(moment(value).format('YYYY-MM-DD 12:12:12'));
          cleared = false;
        }

        if (!value) {
          cleared = true;
        }

        this.setState({ value, cleared });
      }

      render() {
        const { value } = this.state;
        return (
          <DatePicker
            showTime
            value={value}
            format="YYYY-MM-DD HH:mm:ss"
            onChange={this.onChange}
          />
        );
      }
    }

    const wrapper = mount(<Test />);
    // clear input
    clearInput(wrapper);
    openPanel(wrapper);
    selectDate(wrapper, moment('2016-11-13'));
    expect(wrapper.find('.ant-calendar-input').getDOMNode().value).toBe('2016-11-13 12:12:12');
    selectDate(wrapper, moment('2016-11-14'));
    expect(wrapper.find('.ant-calendar-input').getDOMNode().value).toBe('2016-11-14 12:12:12');
  });

  it('triggers onChange only when date was selected', () => {
    const handleChange = jest.fn();
    const wrapper = mount(
      <DatePicker onChange={handleChange} />
    );
    openPanel(wrapper);
    nextYear(wrapper);
    expect(handleChange).not.toBeCalled();
    nextMonth(wrapper);
    expect(handleChange).not.toBeCalled();
    selectDate(wrapper, moment('2017-12-22'));
    expect(handleChange).toBeCalled();
  });

  it('clear input', () => {
    const wrapper = mount(
      <DatePicker />
    );
    openPanel(wrapper);
    selectDate(wrapper, moment('2016-11-23'));
    clearInput(wrapper);
    openPanel(wrapper);
    expect(hasSelected(wrapper, moment('2016-11-22'))).toBe(true);
  });

  it('sets data attributes on input', () => {
    const wrapper = mount(
      <DatePicker data-test="test-id" data-id="12345" />
    );
    const input = wrapper.find('.ant-calendar-picker-input').getDOMNode();
    expect(input.getAttribute('data-test')).toBe('test-id');
    expect(input.getAttribute('data-id')).toBe('12345');
  });

  it('sets aria attributes on input', () => {
    const wrapper = mount(
      <DatePicker aria-label="some-label" aria-labelledby="label-id" />
    );
    const input = wrapper.find('.ant-calendar-picker-input').getDOMNode();
    expect(input.getAttribute('aria-label')).toBe('some-label');
    expect(input.getAttribute('aria-labelledby')).toBe('label-id');
  });

  it('sets role attribute on input', () => {
    const wrapper = mount(
      <DatePicker role="search" />
    );
    const input = wrapper.find('.ant-calendar-picker-input').getDOMNode();
    expect(input.getAttribute('role')).toBe('search');
  });

  it('changes year/month when under control', () => {
    const wrapper = mount(
      <DatePicker value={moment('2018-07-01')} />
    );
    openPanel(wrapper);
    expect(wrapper.find('.ant-calendar-my-select').text()).toBe('Jul2018');
    wrapper.find('.ant-calendar-prev-year-btn').simulate('click');
    wrapper.find('.ant-calendar-prev-month-btn').simulate('click');
    expect(wrapper.find('.ant-calendar-my-select').text()).toBe('Jun2017');
  });
});
