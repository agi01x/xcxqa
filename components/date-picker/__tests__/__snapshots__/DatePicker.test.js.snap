// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker prop locale should works 1`] = `
<span
  class="ant-calendar-picker"
>
  <div
    style=""
  >
    <input
      class="ant-calendar-picker-input ant-input"
      placeholder="Избери дата"
      readonly=""
      value="2000-01-01"
    />
    <i
      class="anticon anticon-close-circle ant-calendar-picker-clear"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="close-circle"
        fill="currentColor"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 0 1-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
        />
      </svg>
    </i>
    <i
      class="anticon anticon-calendar ant-calendar-picker-icon"
    >
      <svg
        aria-hidden="true"
        class=""
        data-icon="calendar"
        fill="currentColor"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
        />
      </svg>
    </i>
  </div>
  <div>
    <div
      class="ant-calendar-picker-container ant-calendar-picker-container-placement-bottomLeft slide-up-appear"
      style="left: -999px; top: -1002px;"
    >
      <div
        class="ant-calendar"
        tabindex="0"
      >
        <div
          class="ant-calendar-panel"
        >
          <div
            class="ant-calendar-input-wrap"
          >
            <div
              class="ant-calendar-date-input-wrap"
            >
              <input
                class="ant-calendar-input "
                placeholder="Избери дата"
                value="2000-01-01"
              />
            </div>
            <a
              role="button"
              title="Изчистване"
            >
              <span
                class="ant-calendar-clear-btn"
              />
            </a>
          </div>
          <div
            class="ant-calendar-date-panel"
          >
            <div
              class="ant-calendar-header"
            >
              <div
                style="position: relative;"
              >
                <a
                  class="ant-calendar-prev-year-btn"
                  role="button"
                  title="Последна година (Control + left)"
                />
                <a
                  class="ant-calendar-prev-month-btn"
                  role="button"
                  title="Предишен месец (PageUp)"
                />
                <span
                  class="ant-calendar-my-select"
                >
                  <a
                    class="ant-calendar-month-select"
                    role="button"
                    title="Избор на месец"
                  >
                    Jan
                  </a>
                  <a
                    class="ant-calendar-year-select"
                    role="button"
                    title="Избор на година"
                  >
                    2000
                  </a>
                </span>
                <a
                  class="ant-calendar-next-month-btn"
                  title="Следващ месец (PageDown)"
                />
                <a
                  class="ant-calendar-next-year-btn"
                  title="Следваща година (Control + right)"
                />
              </div>
            </div>
            <div
              class="ant-calendar-body"
            >
              <table
                cellspacing="0"
                class="ant-calendar-table"
                role="grid"
              >
                <thead>
                  <tr
                    role="row"
                  >
                    <th
                      class="ant-calendar-column-header"
                      role="columnheader"
                      title="Sun"
                    >
                      <span
                        class="ant-calendar-column-header-inner"
                      >
                        Su
                      </span>
                    </th>
                    <th
                      class="ant-calendar-column-header"
                      role="columnheader"
                      title="Mon"
                    >
                      <span
                        class="ant-calendar-column-header-inner"
                      >
                        Mo
                      </span>
                    </th>
                    <th
                      class="ant-calendar-column-header"
                      role="columnheader"
                      title="Tue"
                    >
                      <span
                        class="ant-calendar-column-header-inner"
                      >
                        Tu
                      </span>
                    </th>
                    <th
                      class="ant-calendar-column-header"
                      role="columnheader"
                      title="Wed"
                    >
                      <span
                        class="ant-calendar-column-header-inner"
                      >
                        We
                      </span>
                    </th>
                    <th
                      class="ant-calendar-column-header"
                      role="columnheader"
                      title="Thu"
                    >
                      <span
                        class="ant-calendar-column-header-inner"
                      >
                        Th
                      </span>
                    </th>
                    <th
                      class="ant-calendar-column-header"
                      role="columnheader"
                      title="Fri"
                    >
                      <span
                        class="ant-calendar-column-header-inner"
                      >
                        Fr
                      </span>
                    </th>
                    <th
                      class="ant-calendar-column-header"
                      role="columnheader"
                      title="Sat"
                    >
                      <span
                        class="ant-calendar-column-header-inner"
                      >
                        Sa
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-calendar-tbody"
                >
                  <tr
                    class="ant-calendar-active-week"
                    role="row"
                  >
                    <td
                      class="ant-calendar-cell ant-calendar-last-month-cell"
                      role="gridcell"
                      title="December 26, 1999"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        26
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-last-month-cell"
                      role="gridcell"
                      title="December 27, 1999"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        27
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-last-month-cell"
                      role="gridcell"
                      title="December 28, 1999"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        28
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-last-month-cell"
                      role="gridcell"
                      title="December 29, 1999"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        29
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-last-month-cell"
                      role="gridcell"
                      title="December 30, 1999"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        30
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-last-month-cell ant-calendar-last-day-of-month"
                      role="gridcell"
                      title="December 31, 1999"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        31
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-selected-date ant-calendar-selected-day"
                      role="gridcell"
                      title="January 1, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="true"
                        class="ant-calendar-date"
                      >
                        1
                      </div>
                    </td>
                  </tr>
                  <tr
                    class=""
                    role="row"
                  >
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 2, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        2
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 3, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        3
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 4, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        4
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 5, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        5
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 6, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        6
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 7, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        7
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 8, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        8
                      </div>
                    </td>
                  </tr>
                  <tr
                    class=""
                    role="row"
                  >
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 9, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        9
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 10, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        10
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 11, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        11
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 12, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        12
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 13, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        13
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 14, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        14
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 15, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        15
                      </div>
                    </td>
                  </tr>
                  <tr
                    class=""
                    role="row"
                  >
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 16, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        16
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 17, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        17
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 18, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        18
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 19, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        19
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 20, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        20
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 21, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        21
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 22, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        22
                      </div>
                    </td>
                  </tr>
                  <tr
                    class=""
                    role="row"
                  >
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 23, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        23
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 24, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        24
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 25, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        25
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 26, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        26
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 27, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        27
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 28, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        28
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 29, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        29
                      </div>
                    </td>
                  </tr>
                  <tr
                    class=""
                    role="row"
                  >
                    <td
                      class="ant-calendar-cell"
                      role="gridcell"
                      title="January 30, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        30
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-last-day-of-month"
                      role="gridcell"
                      title="January 31, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        31
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-next-month-btn-day"
                      role="gridcell"
                      title="February 1, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        1
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-next-month-btn-day"
                      role="gridcell"
                      title="February 2, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        2
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-next-month-btn-day"
                      role="gridcell"
                      title="February 3, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        3
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-next-month-btn-day"
                      role="gridcell"
                      title="February 4, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        4
                      </div>
                    </td>
                    <td
                      class="ant-calendar-cell ant-calendar-next-month-btn-day"
                      role="gridcell"
                      title="February 5, 2000"
                    >
                      <div
                        aria-disabled="false"
                        aria-selected="false"
                        class="ant-calendar-date"
                      >
                        5
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              class="ant-calendar-footer"
            >
              <span
                class="ant-calendar-footer-btn"
              >
                <a
                  class="ant-calendar-today-btn "
                  role="button"
                  title="November 22, 2016"
                >
                  Днес
                </a>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</span>
`;
